src/Test/Utilities/Portable/Assert/AssertEx.cs

```diff
@@ -210,6 +214,112 @@ public static void Equal<T>(ImmutableArray<T> expected, ImmutableArray<T> actual
             Assert.True(false, assertMessage);
         }
 
+        /// <summary>
+        /// Asserts that two strings are equal, and prints a diff between the two if they are not.
+        /// </summary>
+        /// <param name="expected">The expected string. This is presented as the "baseline/before" side in the diff.</param>
+        /// <param name="actual">The actual string. This is presented as the changed or "after" side in the diff.</param>
+        /// <param name="message">The message to precede the diff, if the values are not equal.</param>
+        public static void EqualOrDiff(string expected, string actual, string message = null)
+        {
+            if (expected == actual)
+            {
+                return;
+            }
+
+            var diffBuilder = new InlineDiffBuilderWithWhitespace(new Differ());
+            var diff = diffBuilder.BuildDiffModel(expected, actual);
+            var messageBuilder = new StringBuilder();
+            messageBuilder.AppendLine(
+                string.IsNullOrEmpty(message)
+                    ? "Actual and expected values differ. Expected shown in baseline of diff:"
+                    : message);
+
+            foreach (var line in diff.Lines)
+            {
+                switch (line.Type)
+                {
+                    case ChangeType.Inserted:
+                        messageBuilder.Append("+");
+                        break;
+                    case ChangeType.Deleted:
+                        messageBuilder.Append("-");
+                        break;
+                    default:
+                        messageBuilder.Append(" ");
+                        break;
+                }
+
+                messageBuilder.AppendLine(line.Text);
+            }
+
+            Assert.True(false, messageBuilder.ToString());
```

Comment:
I believe the xunit pattern here is just to throw. Otherwise this causes a "Expected: true, actual false" that gets confusing?

Reply:
➡️ I just used the pattern from the previous method.